### PR TITLE
Catch EndOfChannel error and raise TooSlowError instead

### PR DESCRIPTION
**Pull request by njgheorghita**
## What was wrong?
Fixes https://github.com/ethereum/ddht/issues/115

## How was it fixed?

Summary of approach.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/ddht/blob/master/newsfragments/README.md)

[//]: # (See: https://ddht.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/ddht/blob/master/newsfragments/README.md)

#### Cute Animal Picture
image

### CODE REVIEW

File: ddht/v5_1/client.py [hunk 1]
```diff
@@ -405,7 +408,10 @@ async def talk(
             async with self.dispatcher.subscribe_request(
                 request, TalkResponseMessage
             ) as subscription:
-                return await subscription.receive()
+                try:
+                    return await subscription.receive()
+                except trio.EndOfChannel:
+                    raise trio.TooSlowError
```
Comment (pipermerriam) [inline ddht/v5_1/client.py:414]:
```suggestion
                except trio.EndOfChannel as err:
                    raise trio.TooSlowError from err
```

**Review by pipermerriam — COMMENTED** [review]
one more hiding here in the Alexandria part of the codebase: https://github.com/ethereum/ddht/blob/498f1e01cbff44d11d86321add30838a0a82b629/ddht/v5_1/alexandria/client.py#L217-L218

**Review by pipermerriam — APPROVED** [review]
Would you mind tossing in a copy/pasta of the timeout tests we use for the base-client into the alexandria tests so that we are excercising those code paths.  I've been running pretty fast in this codebase and tests aren't as exhaustive as I'd like.

**Comment by pipermerriam** [comment]
(just a test for `AlexandriaClientAPI.ping()` timing out would be adequate.

File: tests/core/v5_1/alexandria/test_alexandria_client.py [hunk 2]
```diff
@@ -131,6 +131,17 @@ async def test_alexandria_client_send_find_nodes(
         assert message.payload.distances == (0, 255, 254)
 
 
+@pytest.mark.trio
```
Comment (njgheorghita) [inline tests/core/v5_1/alexandria/test_alexandria_client.py:134]: @pipermerriam one final :eyes: for this test. This is the only test I could find that seems to test directly exercise the update to `AlexandriaClient.subscribe_request()`. From what I can tell, `AlexandriaClient` only uses the `subscribe_request()` api in `find_nodes()`.

Comment (pipermerriam) [reply tests/core/v5_1/alexandria/test_alexandria_client.py:134]: Correct!  didn't realize that the `AlexandriaClient.ping()` implementation goes directly through the `NetworkAPI.talk()`.  :+1:

File: tests/core/v5_1/alexandria/test_alexandria_client.py [hunk 3]
```diff
@@ -131,6 +131,17 @@ async def test_alexandria_client_send_find_nodes(
...
+async def test_alexandria_client_send_find_nodes_timeout(
+    alice, bob, bob_client, alice_alexandria_client, autojump_clock,
+):
+    with pytest.raises(trio.TooSlowError):
+        async with bob_client.dispatcher.subscribe(TalkRequestMessage) as subscription:
```
Comment (pipermerriam) [inline tests/core/v5_1/alexandria/test_alexandria_client.py:139]: I think you can drop this line, as well as drop out the entire `bob_client` from this test.

**Review by pipermerriam — APPROVED** [review]

**Pull request merged**